metaflow/plugins/kfp/argo_util.py

```diff
@@ -0,0 +1,59 @@
+import shutil
+import subprocess
+import warnings
+
+from kfp.components._yaml_utils import dump_yaml
+
+
+argo_path = shutil.which("argo")
+
+
+def validate_workflow(workflow: dict, kinds: str = "workflows"):
+    if argo_path:
+        has_working_argo_lint = False
+        try:
+            has_working_argo_lint = run_argo_lint(sample_argo_workflow, "workflows")
```

Comment:
This is using `sample_argo_workflow` not `workflow`.

Reply:
This is expected - it first test that argo cli is working as expected. These code are taken from KFP with some modification https://github.com/zillow/pipelines/blob/feature/zg/sdk/python/kfp/compiler/compiler.py#L1235-L1307

Reply:
The actual lint call is on line 24.

Reply:
I see, then it'd be cleaner to inline `sample_argo_workflow` making it easier to read this code.  Especially since `sample_argo_workflow` is only used within this function.